Carbon/components/ChartData.js

```diff
@@ -1,13 +1,118 @@
-
-import { BarChart} from "react-native-chart-kit";
-import {StyleSheet, Dimensions } from 'react-native';
+import React, { useState } from 'react';
+import { View, Text, StyleSheet, Dimensions } from 'react-native';
+import { BarChart } from 'react-native-chart-kit';
+import { VictoryPie } from 'victory-native';
 import { Colors } from '../colors/Colors';
 
 const windowWidth = Dimensions.get("window").width;
 const windowHeight = Dimensions.get("window").height;
 const margin = 10;
 const chartWidth = windowWidth - (margin * 2);
 const chartHeight = 210;
+
+const monthNames = [
+    "January", "February", "March", "April", "May", "June", "July",
+    "August", "September", "October", "November", "December"
+];
+const today = new Date();
+const currentMonth = monthNames[today.getMonth()];
+
+const data = [
+    { x: "Transport", y: Math.round(Math.random() * 10000) },
+    { x: "Diet", y: Math.round(Math.random() * 10000) },
+    { x: "Home", y: Math.round(Math.random() * 10000) },
+    { x: "Stuff", y: Math.round(Math.random() * 10000) },
+];
+
+// Ticket C4-19
+export const CatgegoryChart = () => {
+    const total = data.reduce((acc, datum) => acc + datum.y, 0); //Gets the overall value of all categories
+
+    //Components for the pie chart
+    const pieRadius = (windowWidth - (margin * 2)) * 0.4;
+    const innerRadius = pieRadius * 0.65;
+    const labelRadius = ((pieRadius + innerRadius) / 2);
+
+    // Renders the label for each section of the pie chart
+    const getLabel = (datum) => {
+        const percent = Math.round((datum.y / total) * 100);
+        if (percent > 4) {
+            return `${datum.x}\n${Math.round((datum.y / total) * 100)}%`;
+        }
+        return null;
+    };
+
+    const [selectedSlice, setSelectedSlice] = useState(null);
+
+    const handlePress = (event, props) => {
+        const selectedDatum = data[props.index];
```

Comment:
Please provide comments above these effects so they can be understood what is happening.